src/isomorph.cpp

```diff
@@ -0,0 +1,61 @@
+#include "gap_all.h"
+#include "isomorph.hpp"
+
+
+Obj PermuteMultiplicationTable(Obj self, Obj temp, Obj M, Obj p) {
+    int n = LEN_LIST(M);
```

Comment:
What if `temp` isn't a list? I think you need to check `IS_LIST` and give an error if not.